### DISCUSSION TITLE
PCに接続しましたが認識されません。KEYBALL44。

### DISCUSSION THREAD

**Discussion by etom423** [Yowkees/keyball#517] [Q&A]
当方初心者です。
作り方に従って作業し、テスト用のファームウェアをkeymicroに書き込むところで、PCが認識していない？ようで、ファームウェアの書き込みができない状態になってしまいました。

BOOTHで販売されているkey microを使用しています。コンスルーに対応しているとのことで、はんだ付けは行っておりません。
ケーブルの抜き差し、リセットボタンを押すと接続、切断の音はなるのですが。。。
どなたかご教授いただけると幸いです。
よろしくお願いいたします。

![Wlr17hcO](https://github.com/Yowkees/keyball/assets/164338012/6403eaf9-c0c3-44be-9898-810dc36ce399)

**Comment by koron** [comment]
(issueからdiscussionへ移動しました)

[a bit keysさんのページ](https://abitkeys.com/keymicro/build.html) によりますと、
keymicroのブートローダーはDFUのようです。

> key micro の Bootloader は DFU が書かれています。Firmware を書き込むときは、QMK Toolbox を使って下さい。

一方でKeyball用のコンパイル済みファームウェアは、一般的なPro Micro互換機に合わせて
Caterinaブートローダー(Arduino互換のブートローダー)を指定しています。
そのため仮に書き込めたとしても正しく動作しない可能性があります。

> [!NOTE]
> ブートローダーの違いについての解説はこちらの記事を参照してください。
> [Pro Microとそのバリエーション#ブートローダーの種類](https://zenn.dev/koron/articles/9fee38469a8acc#%E3%83%96%E3%83%BC%E3%83%88%E3%83%AD%E3%83%BC%E3%83%80%E3%83%BC%E3%81%AE%E7%A8%AE%E9%A1%9E)

それでもかまわないということであれば a bit keys さんのページに書いてある通り QMK Toolbox を使ってみてください。
もしくは dfu-programmer というプログラムを利用して書き込むことができるかもしれません。
(QMK Toolboxは正しく設定すれば内部的に dfu-programmer を利用して書き込んだと記憶しています)
なおコンパイル済みファームウェアはCaterina向けにビルドしていますので dfu-programmer で書き込めないかもしれません。

そうである場合はいくつか方法が考えられます。

* 新たにPro Micro互換機を買って、そちらにコンパイル済みファームウェアを書き込んで使う
* ご自身でKeyballのソースコードを DFU ブートローダー向けに変更してビルドしてそちらを使う
* keymicro のブートローダーを Caterina に書き換えてしまう

挙げた順番はオススメ順です。
